### DISCUSSION TITLE
Yellow commit log

### DISCUSSION THREAD

**Discussion by takelley1** [jesseduffield/lazygit#1198] [Q&A]
What does a yellow-colored hash in the commit log signify? It appears that green is for pushed commits and red is for un-pushed commits, but all of my git submodules have yellow hashes (rather than green) in their commit log. Does yellow just mean that the repository is a submodule?

Thanks for the great tool!

**Comment by jesseduffield** [comment]
@takelley1 sorry for late reply, github wasn't notifying me about these threads!

A yellow commit signifies that the commit is not present on the `master` branch, or if you're on a feature branch, that the commit is not present on the `develop` branch. The benefit is you can see how many commits in your branch are actually newly added commits. Perhaps you're not using either of those branches in your repo? If so we could look into making it configurable

**Reply by icorbrey** [reply]
So to clarify: yellow commits are commits not yet present in a branch's parent?

**Reply by stk-ableton** [reply]
Depends on what you mean by "parent". If you mean its base branch, i.e. the one that it was forked off of, and will be merged back into, then yes. If you mean its upstream (i.e. remote tracking) branch, then no.

Also, Jesse's comment above ("If so we could look into making it configurable") is out of date, this _is_ configurable now (with the `git.mainBranches` config).
